src/mon/OSDMonitor.cc

```diff
@@ -799,29 +799,25 @@ void OSDMonitor::maybe_prime_pg_temp()
     }
   } else {
     dout(10) << __func__ << " " << osds.size() << " interesting osds" << dendl;
+    typedef typename mempool::osdmap_mapping::vector<pg_t> acting_pgs_t;
     utime_t stop = ceph_clock_now();
     stop += g_conf->mon_osd_prime_pg_temp_max_time;
-    const int chunk = 1000;
-    int n = chunk;
     std::unordered_set<pg_t> did_pgs;
-    for (auto osd : osds) {
-      auto& pgs = mapping.get_osd_acting_pgs(osd);
+    for (const unsigned osd : osds) {
+      const acting_pgs_t& pgs = mapping.get_osd_acting_pgs(osd);
```

Comment:
I don't understand the benefit of introducing the type here. `auto` has the same effect, except the const.